include/forcing/CsvPerFeatureForcingProvider.hpp

```diff
@@ -301,8 +311,9 @@ class CsvPerFeatureForcingProvider : public forcing::ForcingProvider
 
                 if(well_known_fields.count(n) > 0){
                     auto t = well_known_fields[n];
-                    n = std::get<0>(t);
                     u = u == "" ? std::get<1>(t) : u;
+                    available_forcings_units[n] = u; // Allow lookup of units by non-canonical name
```

Comment:
Yes, I'll improve the variable names, sorry.

Your intuition about the need for an `available_forcings.push_back(n)` is good, but yes this is handled a different way. Since `available_forcings_units` is dimensioned n x 1, I just add a second entry and there is little cost. However, the forcing _data_ is dimensioned n x t, and if I added that with both names, all the data points would have to have a second copy, which seemed memory inefficient. So, the alias is handled by lookup at retrieval time instead within `get_value_for_param_name`.

Reply:
I don't think I follow why you'd need to add a duplicate vector to `forcing_vectors`.   Can't we just  check in `get_value_for_param_name` whether the `name` is a key in `forcing_vectors`, and if not convert it to the "well known" equivalent?

Honestly, I bet we could also adjust `forcing_vectors` to use pointers to the vectors as the map values instead of the vectors directly, and use that to avoid the duplication.

Regardless, I am uneasy about `available_forcings` and `get_available_forcing_outputs` not actually returning the complete collection of valid names for which output can be requested.  I guess that's what I get for not documenting the function, but it seems invalid for something not returned by that to be usable in `get_value` call.  These also feel like assumptions that could easily be erroneously relied upon somewhere else, if they aren't already.

Reply:
>Honestly, I bet we could also adjust forcing_vectors to use pointers to the vectors as the map values instead of the vectors directly, and use that to avoid the duplication.
That was what I was thinking of when I wrote my "Edit" at the end of my above comment.
>Regardless, I am uneasy about available_forcings and get_available_forcing_outputs not actually returning the complete collection of valid names for which output can be requested.
That's tricky to me, in this case...you arguably don't have 14 variables available, you only have 7... at the same time, retrieving a name not in the list "works"... yes, it's somewhat ambiguous.

Reply:
I can take a crack at refactoring that to have `std::shared_ptr<std::vector>`s in the map, but I will definitely want some review, because my "modern C++"-pointer-fu is still weak.

Reply:
>I don't think I follow why you'd need to add a duplicate vector to forcing_vectors. Can't we just check in get_value_for_param_name whether the name is a key in forcing_vectors, and if not convert it to the "well known" equivalent?

Doh... looking at this again I see why you were confused, and I was remembering wrong. `available_forcings` is only the list of names... the values are actually stored in `forcing_vectors`...so yes, we can totally add the name to `available_forcings` as well so that the output of `get_available_forcing_outputs` returns all available names.

Reply:
Fixed in 7020399